packages/purpose-template-process/src/services/validators.ts

```diff
@@ -20,6 +20,9 @@ export const assertConsistentFreeOfCharge = (
   if (isFreeOfCharge && !freeOfChargeReason) {
     throw missingFreeOfChargeReason();
   }
+  if (!isFreeOfCharge && freeOfChargeReason) {
+    throw missingFreeOfChargeReason();
```

Comment:
I don't think it's strictly necessary to add this - we are also diverging from the purpose-process logic. In the comment above, what I suggested was to cover missing cases in the tests.
If you decide to add this error, which can make sense, I think:
- It should be done the same way also in the `purpose-process`
- The error should be different, freeOfChargeReason is present, so it cannot be missingFreeOfChargeReason